### DISCUSSION TITLE
[optimizer] redundant order by elimination

### DISCUSSION THREAD

**Discussion by iwanttobepowerful** [duckdb/duckdb#7837] [Ideas]
for example:
```sql
create table tt1(a int, b int);

explain select count(*) from (
SELECT a from tt1 order by a
);
```

**Comment by hannes** [comment]
Indeed that would be a great idea to have such an optimiser.

**Reply by iwanttobepowerful** [reply]
yes. maybe duckdb can add more rule-based optimizations.

**Reply by iwanttobepowerful** [reply]
https://docs.teradata.com/r/8mHBBLGP88~HK9Auie2QvQ/tJ6ZoFPXf4oUy7lK2vF__g

maybe this is a good reference

**Reply by Tishj** [reply]
Maybe this is a good reference
https://github.com/duckdb/duckdb/tree/master/src/optimizer/rule

Can you clarify what you mean by "duckdb lacks of rule-based optimizations" ?

It can definitely be expanded, but your comment makes it sound like duckdb contains no rule-based optimizations at all

**Reply by iwanttobepowerful** [reply]
yes, you have. take it easy. sorry. But I think you should strive for excellence.
